git_machete/cmd.py

```diff
@@ -1515,6 +1516,84 @@ def flush_caches(self) -> None:
         self.__branch_defs_by_sha_in_reflog = None
         self.__git.flush_caches()
 
+    def checkout_github_pr(self, pr_no: int) -> None:
+        org: str
+        repo: str
+        remote: str
+        remote, (org, repo) = self.__derive_remote_and_github_org_and_repo()
+        debug('checkout_github_pr', f'organization is {org}, repository is {repo}')
+
+        all_prs: List[GitHubPullRequest] = derive_pull_requests(org, repo)
+        pr: GitHubPullRequest = check_pr_already_created_by_number(pr_no, all_prs)
+
+        if not pr:
+            raise MacheteException(f"PR number {pr_no} is not found in repository {repo}")
+        debug('checkout_github_pr', f'found {pr}')
+
+        current_branch: Optional[str] = self.__git.get_currently_checked_out_branch_or_none()
+        if current_branch == 'PR_' + str(pr.number):
+            self.__git.checkout(pr.head)
```

Comment:
Situation when user already checked out given PR, but in the mean time someone added commit into this pull request. This way committed changes are also reflected locally.